### PR TITLE
Search Ichiba Items with given genre Id

### PR DESCRIPTION
**Pull request by satoryu**

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -6,14 +6,18 @@ def index
 
     @items = if @keyword.present?
       RWS::Ichiba::Item.
-        search(keyword: params[:keyword], imageFlag: 1).first(10)
+        search(keyword: params[:keyword], genre_id: @genre.id, imageFlag: 1).first(10)
```
Comment (satoryu) [inline app/controllers/items_controller.rb]: __[rubocop]__
* Line is too long. [86/80]

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -6,14 +6,18 @@ def index
...
     else
       []
     end
   end
 
   private
   def set_genres
-    @genre = RakutenWebService::Ichiba::Genre.root
+    if genre_id = params[:genre_id]
+      @genre = RakutenWebService::Ichiba::Genre.new(genre_id)
```
Comment (satoryu) [inline app/controllers/items_controller.rb]: __[rubocop]__
* Assignment in condition - you probably meant to use `==`.

File: spec/factories/rws_ichiba_genres.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FactoryGirl.define do
+  factory :rws_ichiba_genre, class: "RakutenWebService::Ichiba::Genre" do
+    sequence(:genreId)
+
```
Comment (satoryu) [inline spec/factories/rws_ichiba_genres.rb]: __[rubocop]__
* Trailing whitespace detected.

File: spec/factories/rws_ichiba_genres.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+FactoryGirl.define do
+  factory :rws_ichiba_genre, class: "RakutenWebService::Ichiba::Genre" do
+    sequence(:genreId)
+    
+    skip_create
+    initialize_with { new (attributes) }
```
Comment (satoryu) [inline spec/factories/rws_ichiba_genres.rb]: __[rubocop]__
* `(...)` interpreted as grouped expression.

File: spec/controllers/items_controller_spec.rb [hunk 5]
```diff
@@ -40,5 +42,26 @@
         expect(assigns[:keyword]).to be_eql(keyword)
       end
     end
+
+    context "When given genre_id" do
+      let(:genre) { create(:rws_ichiba_genre, children: []) }
+      let(:keyword) { "foo" }
+      let(:items) { create_list(:rws_ichiba_item, 10) }
+
+      before do
```
Comment (satoryu) [inline spec/controllers/items_controller_spec.rb]: __[rubocop]__
* Trailing whitespace detected.

File: app/controllers/items_controller.rb [hunk 6]
```diff
@@ -6,14 +7,19 @@ def index
 
     @items = if @keyword.present?
       RWS::Ichiba::Item.
-        search(keyword: params[:keyword], imageFlag: 1).first(10)
+        search(keyword: params[:keyword], genre_id: @genre.id, imageFlag: 1).first(10)
```
Comment (mori-sen) [inline app/controllers/items_controller.rb]: __[rubocop]__
* Line is too long. [86/80]

File: spec/factories/rws_ichiba_genres.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+FactoryGirl.define do
```
Comment (mori-sen) [inline spec/factories/rws_ichiba_genres.rb]: __[rubocop]__
* Add an empty line after magic comments. [:ref](https://github.com/bbatsov/ruby-style-guide#separate-magic-comments-from-code)

File: spec/controllers/items_controller_spec.rb [hunk 8]
```diff
@@ -1,9 +1,10 @@
+# frozen_string_literal: true
 require "rails_helper"
 
 RSpec.describe ItemsController, type: :controller do
 
   describe "GET #index" do
-    let(:genre) { double(:genre, children: []) }
+    let(:genre) { create(:rws_ichiba_genre, :root, children: []) }
```
Comment (mori-sen) [inline spec/controllers/items_controller_spec.rb:7]: __[rubocop]__
* ブロックに含まれる行数が多すぎます。 [48/25]

**Pull request merged**